### ISSUE TITLE
Countevals regression

### ISSUE THREAD

**Issue by lxvm** [lxvm/HChebInterp.jl#10]
It seems that v1.1.1 requires more function evaluations in some cases compared to v1.1.0. Here is a mwe
```julia
using HChebInterp

function hchebinterp_count(f, args...; kws...)
    numevals::Int = 0
    g = if f isa BatchFunction
        BatchFunction(f.x) do x
            n += length(x)
            return f.f(x)
        end
    else
        x -> (numevals += 1; f(x))
    end
    fun = hchebinterp(g, args...; kws...)
    return fun, numevals
end

hchebinterp_count(x -> 1e-2/(1e-4+(x-0.42)^2)*1e-2/(1e-4+(x-0.45)^2), 0, 1, rtol=1e-5, atol=1e-5)
```

results with just rtol=1e-5
- v1.1.1: 325 evaluations
- v1.1.0: 298 evaluations

results with just atol=1e-5
- v1.1.1: 493 evaluations
- v1.1.0: 494 evaluations

**Comment by lxvm** [comment]
The change would appear to come from the fact we changed the error tolerance in #9 . [This line](https://github.com/lxvm/HChebInterp.jl/blob/e23fe23109e576fa0fce9c18dd4adc24d74b4352/src/HChebInterp.jl#L193) was changed to use the `maximum(norm, val)` from the parent evaluations (v1.1.0) to the current evaluations (v1.1.1)

**Comment by lxvm** [comment]
The following follow-up test
```julia
f = x -> 1e-2/(1e-4+(x-0.42)^2)*1e-2/(1e-4+(x-0.45)^2)
fun, = hchebinterp_count(f, 0, 1, rtol=1e-5)
rng = range(0, 1, length=3000)
maximum(x -> abs(f(x)-fun(x)), rng) # absolute error
maximum(x -> abs(f(x)-fun(x))/abs(f(x)), rng) # relative error
```
Gives the following results

| version | absolute error | relative error |
| -  | - | - |
| 1.1.1 | 3.806622885349498e-5 | 6.869183148766794e-6 |
| 1.1.0 | 3.806622885349498e-5 | 1.5645472713159266e-5 |

Apparently v1.1.1 requests subdivision of the panel [0.25, 0.375] whereas v1.1.0 does not, and the maximum relative error for v1.1.0 occurs in that panel and exceeds the requested tolerance. Since v1.1.0 is not satisfying the requested tolerance, that is a bug fixed by v1.1.1.
